### PR TITLE
Flaky assertion fix

### PR DESCRIPTION
**Pull request by matejnesuta**
### Describe the change

Related to #7237.

I have not found a reliable way to reproduce the failure, but I have encountered it multiple times both on upstream OCP and Kind. Occasionally, the code inside the Mocha hook was not executed at all and the test passed even though it should detect a failure. This piece of code is tied to most of the CRD validation tests in Cypress.

### Steps to test the PR

Create a minimal Istio gateway in Kiali and insert this scenario in the `istio_config.feature` file:
```
  @bookinfo-app
  @selected
  Scenario: delete me later
    When user selects the "bookinfo" namespace
    Then the "foo" "Gateway" of the "bookinfo" namespace should have a "danger"
```
Run the `yarn cypress:selected` command and the test should fail.

When doing the same procedure against the master branch, there is a chance that the scenario will pass instead of failing.

### Issue reference
#7237

### CODE REVIEW

**Comment by matejnesuta** [comment]
KIA0505 validation is a flaky test, but the other two scenarios are failing consistently on my local Kind.

**Comment by matejnesuta** [comment]
The cleanup hook was not working, so a fix of that will hopefully resolve some of the flakes. KIA1102 is probably a bug (see #7275).

File: frontend/cypress/integration/common/istio_config.ts [hunk 1]
```diff
@@ -522,21 +522,19 @@ Then('the AuthorizationPolicy should have a {string}', function (healthStatus: s
 Then(
   'the {string} {string} of the {string} namespace should have a {string}',
   (crdInstanceName: string, crdName: string, namespace: string, healthStatus: string) => {
-    it('loading config list', { retries: 3 }, () => {
```
[the remark this answers is not in the document]
Comment (matejnesuta) [reply frontend/cypress/integration/common/istio_config.ts:525]: I replaced the original code with my own mechanism using recursion. The cy.wait() interval can be tweaked further.

File: frontend/cypress/integration/common/istio_config.ts [hunk 2]
```diff
@@ -519,31 +519,56 @@ Then('the AuthorizationPolicy should have a {string}', function (healthStatus: s
...
+      if (dataTestAttr !== null) {
+        if (dataTestAttr.value === `VirtualItem_Ns${namespace}_${crdName.toLowerCase()}_${crdInstanceName}`) {
+          found = true;
+        }
+      }
+    })
+    .then(() => {
+      if (!found) {
+        cy.wait(40000);
+        waitUntilConfigIsVisible(attempt - 1, crdInstanceName, crdName, namespace);
```
Comment (ScriptingShrimp) [inline frontend/cypress/integration/common/istio_config.ts:543]: @matejnesuta does the tests eventually stop looping if there is error?

File: frontend/cypress/integration/common/istio_config.ts [hunk 3]
```diff
@@ -519,31 +519,56 @@ Then('the AuthorizationPolicy should have a {string}', function (healthStatus: s
...
+  if (attempt === 0) {
+    return;
+  }
+  cy.request('GET', `${Cypress.config('baseUrl')}/api/istio/config?refresh=0`);
+  cy.get('[data-test="refresh-button"]').click();
+  ensureKialiFinishedLoading();
+  let found = false;
+  cy.get('tr')
+    .each($row => {
+      console.log($row);
```
Comment (hhovsepy) [inline frontend/cypress/integration/common/istio_config.ts:532]: just remove the log, otherwise lgtm, thank you @matejnesuta

**Review by hhovsepy — APPROVED** [review]

File: frontend/cypress/integration/common/istio_config.ts (hunk 2 above)
Comment (matejnesuta) [reply frontend/cypress/integration/common/istio_config.ts:542]: Hello. Yes, there is a condition at the beginning of the function.

Comment (ScriptingShrimp) [reply frontend/cypress/integration/common/istio_config.ts:542]: oh I see now, thanks!

**Review by ScriptingShrimp — COMMENTED** [review]
LGTM

**Pull request merged**